src/vm/checker/read_only/tests.rs

```diff
@@ -50,13 +50,13 @@ fn test_contract_call_read_only_violations() {
             (begin
               (set-entry! tokens (tuple (account tx-sender))
                                               (tuple (balance 10)))
-              'true))";
+              (ok 1)))";
     let bad_caller = 
         "(define-read-only (not-reading-only)
             (contract-call! contract1 mint))";
     let ok_caller =
         "(define-read-only (is-reading-only)
-            (eq? 'null (contract-call! contract1 get-balance)))";
+            (eq? 0 (contract-call! contract1 get-balance)))";
```

Comment:
I don't understand why this is now `0` instead of `'null`?

Reply:
For the purposes of this specific test, it doesn't really matter what the `eq?` argument is -- since this test only is trying to be type-correct, not actually capture some behavior in the contract code. However, this change happened because `'null` is no longer a variable at all.

Reply:
Should type-checking this script fail because `(contract-call! contract1 get-balance` returns an option-type?  I.e. an `int` isn't comparable to an `option<int>`?

Reply:
`eq?` will compare whatever you give it -- if they aren't the same type, it just returns `false`

Reply:
I get a gut feeling that this will lead to some surprising behavior.  For example, would `(eq? 0 (fetch-entry tokens (tuple (account tx-sender)0)` always evaluate to `false` because `fetch-entry` evaluates to `Optional<int>`, which does not have type `int`?

Reply:
Yeah -- I think that's right -- we should typecheck `eq?` arguments

Reply:
Ah, I see you opened an issue on this.  I'll go ahead and resolve this convo here.